qa/setup_packages.py

```diff
@@ -16,34 +16,433 @@
+        """Gets all version strings for with a given cuda_version
+
+            Parameters
+            ----------
+            `cuda_version`: str, optional, default = None
+                Cuda version used for this query
+        """
+        ret = []
+        for i in range(self.get_num_of_version(cuda_version)):
+            ret.append(self.get_version(i, cuda_version))
+        print(self.get_name(cuda_version) + ":\n\t" + "\n\t".join(ret))
+
+class PlainPackage(BasePackage):
+    """Class describing a simple package with a key/name and a list of versions.
+        Cuda version is irrelevant for this package
+
+        Parameters
+        ----------
+        `key`: str
+            Name this package should be queried for
+        `versions`: str
+            List of versions this package is available for
+        `name`: str, , optional, default = None
+            Name of this package used during installation. If empty it is the same as the key
+    """
+    def __init__(self, key, versions, name=None):
+        super(PlainPackage, self).__init__(key, versions, name)
+
+    def get_name(self, cuda_version=None):
+        """Retrives package name.
+
+            Parameters
+            ----------
+            `cuda_version`: str, disregarded, default = None
+                Cuda version used for this query. Disregarded
+        """
+        return self.name
+
+    def get_all_versions(self, cuda_version=None):
+        """Get all versions compatible with provided cuda_version
+
+            Parameters
+            ----------
+            `cuda_version`: str, disregarded, default = None
+                Cuda version used for this query. Disregarded
+        """
+        return self.filter_versions(self.versions)
+
+    def get_install_string(self, idx, cuda_version=None):
+        """Obtains installation string that pip should accept for version at
+            a given index with a given cuda_version
+
+            Parameters
+            ----------
+            `idx`: int
+                Index of version to retrive. If index is beyond 0-num_ov_versions-1 range
+                it is clamped to it
+            `cuda_version`: str, disregarded, default = None
+                Cuda version used for this query. Disregarded
+        """
+        return "{name}=={version}".format(name=self.get_name(cuda_version), version=self.get_version(idx, cuda_version))
+
+    def get_version_string(self, idx, cuda_version=None):
+        """Obtains version string for version at a given index
+            with a given cuda_version
+
+            Parameters
+            ----------
+            `idx`: int
+                Index of version to retrive. If index is beyond 0-num_ov_versions-1 range
+                it is clamped to it
+            `cuda_version`: str, disregarded, default = None
+                Cuda version used for this query. Disregarded
+        """
+        return "{version}".format(version=self.get_version(idx, cuda_version))
```

Comment:
Isn't this a identity function for a string?
What's the difference between get_version_string and get_version?

Reply:
It could be, but you need to supply a cuda version, and that is not possible for `__repr__`.
>What's the difference between get_version_string and get_version

Apparently there is no. Removed.